### ISSUE TITLE
Variable "do" Generated -> JS Error "Unexpected token 'do'"

### ISSUE THREAD

**Issue by noblemaster** [konsoletyper/teavm#980]
The generated JavaScript file produces a variable `do`, preveting the JavaScript application to run at all. This is for TeaVM version 0.10.2. 

Generally, when a `dn` is generated, the next generated name is `dp` which is fine. However, there were a few places where instead the `do` was created.

Examples from the generated .js File (I can provide more as needed):
```
let JEf=(a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q,r,s,t,u,v,w,x,y,z,ba,bb,bc,bd,be,bf,bg,bh,bi,bj,bk,bl,bm,bn,bo,bp,bq,br,bs,bt,bu,bv,bw,bx,by,bz,bA,bB,bC,bD,bE,bF,bG,bH,bI,bJ,bK,bL,bM,bN,bO,bP,bQ,bR,bS,bT,bU,bV,bW,bX,bY,bZ,b0,b1,b2,b3,b4,b5,b6,b7,b8,b9,b$,b_,ca,cb,cc,cd,ce,cf,cg,ch,ci,cj,ck,cl,cm,cn,co,cp,cq,cr,cs,ct,cu,cv,cw,cx,cy,cz,cA,cB,cC,cD,cE,cF,cG,cH,cI,cJ,cK,cL,cM,cN,cO,cP,cQ,cR,cS,cT,cU,cV,cW,cX,cY,cZ,c0,c1,c2,c3,c4,c5,c6,c7,c8,c9,c$,c_,da,db,dc,dd,de,df,dg,dh,di,dj,dk,dl,dm,dn,dp,dq,dr)=>{a.iwa=b;a.geY
=c;a.gY2=d;a.iXC=e;a.gCP=f;a.ic2=g;a.h8T=h;a.itp=i;a.ine=j;a.kLU=k;a.kIT=l;a.kk$=m;a.kR1=n;a.jWO=o;a.kpU=p;a.k$r=q;a.kmX=r;a.lba=s;a.jZ9=t;a.jYh=u;a.kww=v;a.jQ0=w;a.j_b=x;a.kXS=y;a.k8S=z;a.kH_=ba;a.kDX=bb;a.kW1=bc;a.j$z=bd;a.kJE=be;a.ku4=bf;a.jZ_=bg;a.j5i=bh;a.k5W=bi;a.jVW=bj;a.kcN=bk;a.ktC=bl;a.kna=bm;a.kvb=bn;a.kem=bo;a.kQK=bp;a.kk6=bq;a.kUP=br;a.j5X=bs;a.j3j=bt;a.kw_=bu;a.kGA=bv;a.k1D=bw;a.kuT=bx;a.laO=by;a.j5m=bz;a.kQb=bA;a.kPh=bB;a.kL8=bC;a.klf=bD;a.kdS=bE;a.kgQ=bF;a.kd0=bG;a.jWt=bH;a.kYL=bI;a.kAK=bJ;a.kyd
=bK;a.kuV=bL;a.kJr=bM;a.kOU=bN;a.jYd=bO;a.kds=bP;a.ke8=bQ;a.kXV=bR;a.jTw=bS;a.kO_=bT;a.kRt=bU;a.jZI=bV;a.kYx=bW;a.je_=bX;a.kTe=bY;a.kTB=bZ;a.k39=b0;a.kkL=b1;a.k4v=b2;a.ktb=b3;a.kD_=b4;a.j2g=b5;a.koR=b6;a.j9W=b7;a.kTa=b8;a.jBz=b9;a.kyG=b$;a.j4r=b_;a.koT=ca;a.kaF=cb;a.ktV=cc;a.kOF=cd;a.kHp=ce;a.kt5=cf;a.kdb=cg;a.kU8=ch;a.ko_=ci;a.kVn=cj;a.kO6=ck;a.j6e=cl;a.ldK=cm;a.kH1=cn;a.kA5=co;a.kXb=cp;a.kh4=cq;a.k6G=cr;a.kSt=cs;a.kIN=ct;a.k62=cu;a.kCZ=cv;a.kyU=cw;a.jXy=cx;a.kwS=cy;a.k5s=cz;a.kqx=cA;a.j6_=cB;a.kiB=cC;a.knJ
=cD;a.kMx=cE;a.kMy=cF;a.kls=cG;a.j$k=cH;a.kPR=cI;a.kGq=cJ;a.kGr=cK;a.kGo=cL;a.kGp=cM;a.kpI=cN;a.jRp=cO;a.kt1=cP;a.j0v=cQ;a.j0u=cR;a.kZX=cS;a.kEu=cT;a.kEv=cU;a.kEt=cV;a.kTw=cW;a.kpK=cX;a.la4=cY;a.kti=cZ;a.kiw=c0;a.kiv=c1;a.kix=c2;a.ldX=c3;a.kyz=c4;a.kRh=c5;a.kRg=c6;a.kfo=c7;a.k6V=c8;a.lcg=c9;a.kyi=c$;a.j$K=c_;a.kVN=da;a.kUA=db;a.kmq=dc;a.kVK=dd;a.kQX=de;a.kws=df;a.klu=dg;a.kr5=dh;a.lbV=di;a.j_e=dj;a.kTX=dk;a.jYq=dl;a.kmD=dm;a.kqi=dn;a.kqv=dp;a.j0y=dq;a.ld3=dr;DK(a);},Nom=(a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q,r,
s,t,u,v,w,x,y,z,ba,bb,bc,bd,be,bf,bg,bh,bi,bj,bk,bl,bm,bn,bo,bp,bq,br,bs,bt,bu,bv,bw,bx,by,bz,bA,bB,bC,bD,bE,bF,bG,bH,bI,bJ,bK,bL,bM,bN,bO,bP,bQ,bR,bS,bT,bU,bV,bW,bX,bY,bZ,b0,b1,b2,b3,b4,b5,b6,b7,b8,b9,b$,b_,ca,cb,cc,cd,ce,cf,cg,ch,ci,cj,ck,cl,cm,cn,co,cp,cq,cr,cs,ct,cu,cv,cw,cx,cy,cz,cA,cB,cC,cD,cE,cF,cG,cH,cI,cJ,cK,cL,cM,cN,cO,cP,cQ,cR,cS,cT,cU,cV,cW,cX,cY,cZ,c0,c1,c2,c3,c4,c5,c6,c7,c8,c9,c$,c_,da,db,dc,dd,de,df,dg,dh,di,dj,dk,dl,dm,dn,do,dp)=>{let dq=new KGo();JEf(dq,a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q,r,s,
t,u,v,w,x,y,z,ba,bb,bc,bd,be,bf,bg,bh,bi,bj,bk,bl,bm,bn,bo,bp,bq,br,bs,bt,bu,bv,bw,bx,by,bz,bA,bB,bC,bD,bE,bF,bG,bH,bI,bJ,bK,bL,bM,bN,bO,bP,bQ,bR,bS,bT,bU,bV,bW,bX,bY,bZ,b0,b1,b2,b3,b4,b5,b6,b7,b8,b9,b$,b_,ca,cb,cc,cd,ce,cf,cg,ch,ci,cj,ck,cl,cm,cn,co,cp,cq,cr,cs,ct,cu,cv,cw,cx,cy,cz,cA,cB,cC,cD,cE,cF,cG,cH,cI,cJ,cK,cL,cM,cN,cO,cP,cQ,cR,cS,cT,cU,cV,cW,cX,cY,cZ,c0,c1,c2,c3,c4,c5,c6,c7,c8,c9,c$,c_,da,db,dc,dd,de,df,dg,dh,di,dj,dk,dl,dm,dn,do,dp);return dq;},Qwg=(a,b)=>{let c,$p,$z;$p=0;if(H()){let $T=F();$p=$T.l();c
```

```
DC([KGo,0,Cj,[],0,0,0,0,["d70",function(b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q,r,s,t,u,v,w,x,y,z,ba,bb,bc,bd,be,bf,bg,bh,bi,bj,bk,bl,bm,bn,bo,bp,bq,br,bs,bt,bu,bv,bw,bx,by,bz,bA,bB,bC,bD,bE,bF,bG,bH,bI,bJ,bK,bL,bM,bN,bO,bP,bQ,bR,bS,bT,bU,bV,bW,bX,bY,bZ,b0,b1,b2,b3,b4,b5,b6,b7,b8,b9,b$,b_,ca,cb,cc,cd,ce,cf,cg,ch,ci,cj,ck,cl,cm,cn,co,cp,cq,cr,cs,ct,cu,cv,cw,cx,cy,cz,cA,cB,cC,cD,cE,cF,cG,cH,cI,cJ,cK,cL,cM,cN,cO,cP,cQ,cR,cS,cT,cU,cV,cW,cX,cY,cZ,c0,c1,c2,c3,c4,c5,c6,c7,c8,c9,c$,c_,da,db,dc,dd,de,df,dg,dh,di,dj,dk,dl,dm,
dn,do,dp,dq){JEf(this,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q,r,s,t,u,v,w,x,y,z,ba,bb,bc,bd,be,bf,bg,bh,bi,bj,bk,bl,bm,bn,bo,bp,bq,br,bs,bt,bu,bv,bw,bx,by,bz,bA,bB,bC,bD,bE,bF,bG,bH,bI,bJ,bK,bL,bM,bN,bO,bP,bQ,bR,bS,bT,bU,bV,bW,bX,bY,bZ,b0,b1,b2,b3,b4,b5,b6,b7,b8,b9,b$,b_,ca,cb,cc,cd,ce,cf,cg,ch,ci,cj,ck,cl,cm,cn,co,cp,cq,cr,cs,ct,cu,cv,cw,cx,cy,cz,cA,cB,cC,cD,cE,cF,cG,cH,cI,cJ,cK,cL,cM,cN,cO,cP,cQ,cR,cS,cT,cU,cV,cW,cX,cY,cZ,c0,c1,c2,c3,c4,c5,c6,c7,c8,c9,c$,c_,da,db,dc,dd,de,df,dg,dh,di,dj,dk,dl,dm,dn,do,dp,dq);},"cW",
```

**Comment by konsoletyper** [comment]
Hm, that's not enough issue for me. Is it related to local variables?

**Comment by noblemaster** [comment]
Not sure? The generated obfuscated code produces a `do` within the parameters/variables. The `do` then prevents the JavaScript application to run.

**Comment by noblemaster** [comment]
![image](https://github.com/user-attachments/assets/138702e0-e97a-4543-90c7-0be57f05f9df)

**Comment by konsoletyper** [comment]
So, it's a function parameter? I see. But how come you have function with such huge amount of parameters?

**Comment by noblemaster** [comment]
I don't have a function with that many parameters. Maybe it's an enum/etc. that got converted to a function? It's obfuscated, thus difficult to tell where it all comes from?

**Comment by konsoletyper** [comment]
@noblemaster there's always the way to map obfuscated method back to original sources. For example, you can generate source map or TeaVM debug info. In first case you can use any source map tool, in second there's no ready-to-use solution, but you can write a simple tool. Just include `teavm-core` and use `org.teavm.debugging.information.DebugInformation` class (to obtain its instance, use `read` static method). For example, call `getMethodAt`.

**Comment by konsoletyper** [comment]
> Maybe it's an enum/etc. that got converted to a function?

Doubtful. Enums don't produce constructors and/or virtual methods with 100 parameters. Perhaps, it's something that generates libGDB backend internally?

**Comment by konsoletyper** [comment]
@noblemaster I just committed possible fix for this issue. Can you please check? Should I assist you with building TeaVM locally?

**Comment by noblemaster** [comment]
Thanks for the fix! I created a local TeaVM 0.11.0-SNAPSHOT release and built the project again using that SNAPSHOT. In any case, I was able to get the project to succesfully compile without the `do` keyword! Thank you, **that resolves the issue**. 😄 

When do you think can you publish another release version (no rush!)?
